timemachine/maps/terminal_bonds.py

```diff
@@ -0,0 +1,236 @@
+    bonds = [sort_by_descending_degree(bond) for bond in bond_idxs]
+
+    is_terminal = lambda bond: degrees[bond[1]] == 1
+
+    terminal_bonds = sorted(filter(is_terminal, bonds))
+    return terminal_bonds
+
+
+class TerminalMappableState:
+    def __init__(self, terminal_bond_idxs, ks, eq_lengths, temperature=DEFAULT_TEMP, sigma_thresh=20):
+        """
+        for each (a, b) in terminal_bond_idxs,
+        prepare to construct a map that moves b -> b', conditioned on a
+
+        Parameters
+        -----------
+        terminal_bond_idxs: [B,2] ints
+            bonds that will be mapped,
+            assumed in order (anchor, terminal)
+
+        ks: [B,] positive floats
+            force constants of terminal bonds
+        eq_lengths: [B,] positive floats
+            equilibrium lengths of terminal bonds
+
+        temperature: positive float, assumed in kelvin
+            determines the stddev of the modeled bond length distribution
+            bond_length_distribution = Gaussian(mu=eq_length, sigma=sqrt(BOLTZ * temperature / k))
+
+        sigma_thresh: positive float, unitless
+            determines the size of the Interval mu +/- sigma_thresh * sigma to which the support of
+            bond_length_distribution will be truncated
+        """
+
+        B = len(terminal_bond_idxs)
+        assert (len(ks) == B) and (len(eq_lengths) == B)
+
+        self.idxs = terminal_bond_idxs
+        self.ks = ks
+        self.eq_lengths = eq_lengths
+
+        self.temperature = temperature
+        self.sigma_thresh = sigma_thresh
+
+        self.gaussians = [Gaussian.from_harmonic_bond(k, r0, temperature) for k, r0 in zip(ks, eq_lengths)]
+        self.intervals = [g.to_interval(sigma_thresh) for g in self.gaussians]
+
+    def contains_in_support(self, x) -> bool:
+        """returns whether x is in the support of state"""
+
+        bond_valid = []
+
+        for i in range(len(self.idxs)):
+            a, b = self.idxs[i]
+
+            r = jnp.linalg.norm(x[a] - x[b])
+            interval = self.intervals[i]
+            bond_valid.append((r <= interval.upper) * (r >= interval.lower))
+
+        return jnp.array(bond_valid).all()
+
+    @classmethod
+    def from_harmonic_bond_params(cls, bond_idxs, params, temperature=DEFAULT_TEMP, sigma_thresh=20):
+        # bond (i, j) -> (k, eq_length)
+        param_dict = dict(zip(map(tuple, bond_idxs), params))  # bond idxs may be in arbitrary order
+
+        terminal_bond_tuples = get_terminal_bonds(bond_idxs)  # each tuple will be sorted in order (anchor, terminal)
+
+        ks, eq_lengths = np.array([param_dict[tuple(sorted(bond))] for bond in terminal_bond_tuples]).T
+
+        return cls(np.array(terminal_bond_tuples), ks, eq_lengths, temperature=temperature, sigma_thresh=sigma_thresh)
+
+
+def states_to_conf_map_params(src: TerminalMappableState, dst: TerminalMappableState):
+    # find bond idxs in common
```

Comment:
Can you add a brief summary of this function?

Reply:
Added docstring in https://github.com/proteneer/timemachine/pull/801/commits/d4339076819ebb2aaf464babb646e0b9a5199b6c .

Note that you don't need to call this function directly -- intended usage is [`TerminalBondMap.from_states(src, dst)`](https://github.com/proteneer/timemachine/blob/d4339076819ebb2aaf464babb646e0b9a5199b6c/timemachine/maps/terminal_bonds.py#L265) , as [in the tests](https://github.com/proteneer/timemachine/blob/d4339076819ebb2aaf464babb646e0b9a5199b6c/tests/test_terminal_bond_maps.py#L122-L124), but figured I should expose the implementation as a free function as well, for ease of tinkering.